File: trs-ui/src/app/reservation/model/reservation.ts
import {FlightDetails} from "../../flight/model/flight-details";
import {Customer} from "../../customer/model/customer";
import {FlightDepartureTime} from "../../flight/model/flight-departure-time";

export class Reservation {
  id: bigint | undefined;
  customer: Customer | undefined;
  flightDetails: FlightDetails | undefined;
  flightDepartureTime: FlightDepartureTime | undefined;
  reservationDate: Date | undefined;
  paid: boolean | undefined;
}

File: TRS-Backend/src/main/java/com/airplane/TicketReservationSystem/entity/Reservation.java
package com.airplane.TicketReservationSystem.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
public class Reservation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "customer_id", referencedColumnName = "id")
    private Customer customer;

    @ManyToOne
    @JoinColumn(name = "flight_id", referencedColumnName = "id")
    private FlightDetails flightDetails;

    @ManyToOne
    @JoinColumn(name="departure_time_id", referencedColumnName = "id")
    private FlightDepartureTime flightDepartureTime;

    private Date reservationDate;

    private boolean isPaid;

    @OneToOne(mappedBy = "reservation")
    @JsonIgnore
    private Payment payment;
}

